src/utils/automation-helper.js

```diff
@@ -29,10 +29,337 @@ export const getAutomation = async (automationId) => {
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.AutomationError'), ok: true };
+      }
+      break;
+    }
+    case 'finished with warnings': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+      break;
+    }
+    case 'must stop':
+    case 'stopped': {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = false;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.UnkownError'), ok: true };
+      }
+      break;
+    }
+    default: {
+      if (finalRun.title?.length > 0) {
+        msg = parseOutput(finalRun.title, translator);
+        msg.ok = true;
+      } else {
+        msg = { message: translator.get('Object.ActionButton.Automation.DefaultAutomationMsg'), ok: true };
+      }
+    }
+  }
+  return msg;
+};
+
+export const getAutomationMsg = async (automationId, triggered, response, translator) => {
+  let message;
+  switch (response.status) {
+    case 200:
+    case 201: {
+      const data = await response.json();
+      const { status, guid, id } = data;
+      const queued = status === 'queued';
+      const runId = typeof id === 'undefined' ? guid : id;
+      if (!triggered || queued) {
+        message = automationRunPolling(automationId, runId, translator);
+      } else {
+        message = parseOutput(data, translator);
+        message.ok = true;
+      }
+      break;
+    }
+    case 400: {
+      message = { message: translator.get('Object.ActionButton.Automation.BadRequest'), ok: false };
+      break;
+    }
+    case 401:
+    case 403: {
+      message = { message: translator.get('Object.ActionButton.Automation.NotAuthorized'), ok: false };
+      break;
+    }
+    case 404: {
+      message = { message: translator.get('Object.ActionButton.Automation.AutomationNotFound'), ok: false };
+      break;
+    }
+    case 500:
+    case 503: {
+      message = { message: translator.get('Object.ActionButton.Automation.AutomationError'), ok: false };
+      break;
+    }
+    default: {
+      message = { message: translator.get('Object.ActionButton.Automation.UnkownError'), ok: false };
+    }
+  }
+  return message;
+};
+
+const applyStyles = (element, styles) => {
+  if (element !== undefined && element !== null) {
+    Object.keys(styles).forEach((key) => {
+      element.style[key] = styles[key];
+    });
+  }
+};
+
+const getSnackbar = (snackbarId) => document.querySelector(`#${snackbarId}`);
+
+const removeSnackbar = (element) => {
+  applyStyles(element, { opacity: 0 });
+  setTimeout(() => {
+    element.remove();
+  }, 1000);
+};
+
+export const createSnackbar = async (message, duration, newTab) => {
```

Comment:
I tried using a HTML template as this would be far easier to see what will be created in the end but I had problems importing it (most likely we would need to touch the build process to allow html files to be imported etc) - Therefore I tried to define the HTML as a string and use less code to make it a bit more readable. Another idea would be to bring React in, however might be a bit overkill since we only need this one static snackbar?